### PR TITLE
Add a new fromBytes that doesn't ignore errors

### PR DESCRIPTION
**Pull request by OceanOak**
No changelog

This adds a new fromBytes that doesn't ignore errors and uses it in String.base64Decode

### CODE REVIEW

File: backend/src/BuiltinExecution/Libs/String.fs [hunk 1]
```diff
@@ -421,6 +421,26 @@ let fns : List<BuiltInFn> =
       deprecated = NotDeprecated }
 
 
+    { name = fn "fromBytesRecoverable" 0
```
Comment (pbiggar) [inline backend/src/BuiltinExecution/Libs/String.fs:424]: I think this is backwards.

Errors will be ignored in the function that returns a String, and errors will not be ignored in the function that returns `Option<String>`.

I had intended fromBytesRecoverable to be the name for the function that returned a string (and hence recovers). But I think it's a poor name.

Comment (OceanOak) [reply backend/src/BuiltinExecution/Libs/String.fs:424]: Please correct me if I miss-understood what you mean here:
- The original fromBytes we had (which returns a String) should be renamed to something that indicates that it recovers)
- The newly added fromBytes (currently named fromBytesRecoverable) should be renamed to fromBytes
- We should then update all the references to the original 'fromBytes' method with its new name

Name suggestions:
- fromBytesWithReplacement
- fromBytesToSafeString
- fromBytesIgnoreErrors

Comment (pbiggar) [reply backend/src/BuiltinExecution/Libs/String.fs]: I like `fromBytesWithReplacement`

Yes, this sounds right. The default should be that the Bytes might not be strings and the user should opt into replacement.

File: backend/testfiles/execution/stdlib/string.dark [hunk 2]
```diff
@@ -252,15 +252,14 @@ module Base64Decode =
     "👱👱🏻👱🏼👱🏽👱🏾👱🏿"
 
 
-  // These produce strings of bytes which are technically legal it seems
-  PACKAGE.Darklang.Stdlib.String.base64Decode_v0 "-p" = PACKAGE.Darklang.Stdlib.Result.Result.Ok
-    "�"
```
Comment (pbiggar) [inline backend/testfiles/execution/stdlib/string.dark:257]: Can you add tests for fromBytesWithReplacement?

**Review by pbiggar — APPROVED** [review]
One note for followup.

**Pull request merged**